client/components/domains/use-your-domain-step/index.tsx

```diff
@@ -0,0 +1,311 @@
+	image,
+	title,
+	reasons,
+	onClick,
+	buttonText,
+	isPrimary,
+	learnMore,
+}: OptionContentProps ) {
+	return (
+		<Card className="use-your-domain-step__option" compact>
+			<div className="use-your-domain-step__option-inner-wrap">
+				<div className="use-your-domain-step__option-content">
+					<div className="use-your-domain-step__option-illustration">
+						<img src={ image } alt="" />
+					</div>
+					<h3 className="use-your-domain-step__option-title">{ title }</h3>;
+					<div className="use-your-domain-step__option-reasons">
+						{ reasons.map( ( phrase, index ) => {
+							if ( isEmpty( phrase ) ) {
+								return;
+							}
+
+							return (
+								<div className="use-your-domain-step__option-reason" key={ index }>
+									<Gridicon icon="checkmark" size={ 18 } />
+									{ phrase }
+								</div>
+							);
+						} ) }
+					</div>
+				</div>
+				<div className="use-your-domain-step__option-action">
+					<Button
+						className="use-your-domain-step__option-button"
+						primary={ isPrimary }
+						onClick={ onClick }
+					>
+						{ buttonText }
+					</Button>
+					<div className="use-your-domain-step__learn-more">{ learnMore }</div>
+				</div>
+			</div>
+		</Card>
+	);
+}
+
+type UseYourDomainStepProps = {
+	basePath: string;
+	goBack?: () => void;
+	initialQuery?: string;
+};
+
+function UseYourDomainStep( { basePath, goBack, initialQuery }: UseYourDomainStepProps ) {
+	const cartKey = useCartKey();
+	const { responseCart: cart } = useShoppingCart( cartKey );
+
+	const translate = useTranslate();
+
+	const [ searchQuery ] = useState( initialQuery || '' );
+
+	const selectedSite = useSelector( getSelectedSite );
+	const productsList = useSelector( getProductsList );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const domainsWithPlansOnly = useSelector( ( state ) =>
+		currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+	);
+	const primaryWithPlansOnly = useSelector( ( state ) =>
+		currentUserHasFlag( state, NON_PRIMARY_DOMAINS_TO_FREE_USERS )
+	);
+
+	const domainsWithPlansOnlyButNoPlan =
+		domainsWithPlansOnly && selectedSite?.plan && ! isPlan( selectedSite.plan );
+	const productSlug = getDomainProductSlug( searchQuery );
+	const domainProductSalePrice = getDomainTransferSalePrice(
+		productSlug,
+		productsList,
+		currencyCode
+	);
+
+	let domainProductFreeText = null;
+	if ( isNextDomainFree( cart ) || isDomainBundledWithPlan( cart, searchQuery ) ) {
+		domainProductFreeText = translate( 'Free with your plan' );
+	} else if ( domainsWithPlansOnlyButNoPlan ) {
+		domainProductFreeText = translate( 'Included in paid plans' );
+	}
+
+	let transferSalePriceText = null;
+	if (
+		! isEmpty( domainProductSalePrice ) &&
+		! isNextDomainFree( cart ) &&
+		! isDomainBundledWithPlan( cart, searchQuery ) &&
+		domainsWithPlansOnlyButNoPlan
+	) {
+		transferSalePriceText = translate( 'Sale price is %(cost)s', {
+			args: { cost: domainProductSalePrice! },
```

Comment:
Added a `!` because `isEmpty` is not acting as a type guard

Reply:
You can remove the `isEmpty` check (it's lodash, so it's unwanted anyway) and replace it with `domainProductSalePrice !== null`.

Reply:
https://github.com/Automattic/wp-calypso/pull/101854/commits/f52f737944224af3059dfe29f5ad3d8ff3c5dcdb